src/vm.rs

```diff
@@ -172,6 +172,12 @@ impl SyscallRegistry {
     pub fn get_number_of_syscalls(&self) -> usize {
         self.entries.len()
     }
+
+    /// Calculate memory size
+    pub fn mem_size(&self) -> usize {
+        self.entries.len() * (std::mem::size_of::<u32>() + std::mem::size_of::<Syscall>())
```

Comment:
inconsistent style, I would prefer using the tuple style here as well: `std::mem::size_of::<(u32, Syscall)>()`

Reply:
Also `+ std::mem::size_of::<Self>()`?

Reply:
Lots of room for improvement.  Not sure how maintainable even a good implementation would be, so just trying to get a rough estimate for now.  Needs lots of honing before we would check this in